### PR TITLE
Added data source to get Kafka topic config

### PR DESCRIPTION
**Pull request by manjeetvk**
Added terraform data source to get Kafka topic config.

### CODE REVIEW

File: kafka/provider.go [hunk 1]
```diff
@@ -114,6 +114,9 @@ func Provider() terraform.ResourceProvider {
 			"kafka_topic": kafkaTopicResource(),
 			"kafka_acl":   kafkaACLResource(),
 		},
+		DataSourcesMap: map[string]*schema.Resource{
+			"kafka_topic_config": kafkaTopicConfigDataSource(),
```
Comment (llamahunter) [inline kafka/provider.go:118]: shouldn't this be called `kafka_topic`?  One of the subfields is the actual config.

File: kafka/data_source_kafka_topic.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package kafka
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func kafkaTopicConfigDataSource() *schema.Resource {
```
Comment (llamahunter) [inline kafka/data_source_kafka_topic.go:9]: More appropriately named `kafkaTopicDataSource`, no?

File: kafka/data_source_kafka_topic.go [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+				Type:         schema.TypeInt,
+				Computed:     true,
+				Description:  "Number of partitions.",
+			},
+			"replication_factor": {
+				Type:         schema.TypeInt,
+				Computed:     true,
+				Description:  "Number of replicas.",
+			},
+			"config": {
```
Comment (llamahunter) [inline kafka/data_source_kafka_topic.go:28]: the `config` is an embedded field

**Review by llamahunter — COMMENTED** [review]
I think the name of the data source should be just `kafka_topic` to match the corresponding resource that it reads.  That seems to be the general convention elsewhere in terraform providers.

**Comment by manjeetvk** [comment]
@Mongey Any thoughts on this new data source.

File: kafka/provider.go (hunk 1 above)
Comment (Mongey) [reply kafka/provider.go:118]: @manjeetvk yeah, can we rename this to `kafka_topic`

File: kafka/data_source_kafka_topic.go [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func kafkaTopicConfigDataSource() *schema.Resource {
+	return &schema.Resource{
+		Read:   topicConfigRead,
```
Comment (Mongey) [inline kafka/data_source_kafka_topic.go:11]: We could reuse the `topicRead` function

File: kafka/data_source_kafka_topic.go [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+		Schema: map[string]*schema.Schema{
```
Comment (Mongey) [inline kafka/data_source_kafka_topic.go:12]: Could potentially reuse the topic schema here.

**Review by Mongey — CHANGES_REQUESTED** [review]

**Comment by llamahunter** [comment]
I think this is superseded by #168

**Pull request closed without merging**